### PR TITLE
Document that State#build(BuildContext context) is always the same as State#context

### LINKED ISSUE #4114: Document that State#build(BuildContext context) is always the same as State#context

### PR DESCRIPTION
**Pull request by abarth**
Fixes #4114

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -461,6 +461,10 @@ abstract class State<T extends StatefulWidget> {
   /// The given build context object contains information about the location in
   /// the tree at which this widget is being built. For example, the context
   /// provides the set of inherited widgets for this location in the tree.
+  ///
+  /// The context argument is always the same as [context]. This argument is
```
Comment (a14n) [inline packages/flutter/lib/src/widgets/framework.dart]: `[context]` should be `[State.context]` to reference the getter instead of param.

Comment (abarth) [inline packages/flutter/lib/src/widgets/framework.dart]: Thanks, fixed.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**